lib/TelemetryFeed.jsm

```diff
@@ -556,6 +568,32 @@ this.TelemetryFeed = class TelemetryFeed {
     }
   }
 
+  /**
+   * Generates an endpoint for Structured Ingestion telemetry pipeline. Note that
+   * Structured Ingestion requires a different endpoint for each ping. See more
+   * details about endpoint schema at:
+   * https://github.com/mozilla/gcp-ingestion/blob/master/docs/edge.md#postput-request
+   *
+   * @param {String} pingType  Type of the ping, such as "impression-stats".
+   * @param {String} version   Endpoint version for this ping type.
+   */
+  _generateStructuredIngestionEndpoint(pingType, version) {
+    const uuid = gUUIDGenerator.generateUUID().toString();
+    // Structured Ingestion does not support the UUID generated by gUUIDGenerator,
+    // because it contains leading and trailing braces. Need to trim them first.
+    const docID = uuid.substring(1, uuid.length - 1);
```

Comment:
You can also do `uuid.slice(1, -1)` but I guess that's pretty much the same minus checking the length.

Reply:
Good idea, let's stick to slice.